### DISCUSSION TITLE
"pac connector list" command not working

### DISCUSSION THREAD

**Discussion by joes** [microsoft/powerplatform-build-tools#887] [Ideas]
I am connected to an environment and am not able to list the custom connectors there:

```console
pac.exe connector list
Ansluten till ... The Environment
Ansluten som someone@example.com
Det finns inga anslutningsprogram i miljön.
```

It tells me in Swedish (since I cannot change the language) that there were no connectors in the environment. However, there is one (1) custom connector in said environment.

Now, I am assuming that "connectors" == "custom connectors" since the `create` subcommand has all the options compatible with creating such connectors:

```
Usage:  pac connector create [--api-definition-file] [--api-properties-file] [--icon-file] [--script-file] [--solution-unique-name] [--environment] [--settings-file]
```

Compare with the tool [paconn](https://learn.microsoft.com/en-us/connectors/custom-connectors/paconn-cli) which is able to list the connector in this environment:

```console
$ paconn download
Please select an environment:
 [1] The Environment
 [2] Default env (default)
Please enter a choice [Default choice(1)]: 1
Environment selected: The Environment
Please select a connector:
 [1] The connector - John Doe
```

However, `paconn` has no command to list connectors. It would be great to be able to use one command (i.e. pac.exe) for everything related to custom connector management.

**Comment by joes** [comment]
It seems `pac.exe` is not aware of connectors in the defalt solution whereas `paconn` is aware of connectors in both the default solution and "custom" solutions. That is, `pac.exe` is only aware of custom connectors created from the context of a "custom" (non-default) solution. 

It is impossible to see if a connector is packed within a solution or not in the "Custom connectors" menu in Power Automate - they will look the same.

**Comment by Edward-Zhou** [comment]
There are two source for custom connector, `From Dataverse` and `Outside Dataverse`. For `pac.exe`, it seems can only list connector under dataverse.
Is it possible to list all connectors?
